libats/src/main/scala/com/advancedtelematic/libats/data/EcuIdentifier.scala

```diff
@@ -9,13 +9,14 @@ object EcuIdentifier {
 
   implicit val validatedEcuIdentifier = new ValidatedGeneric[EcuIdentifier, String] {
     override def to(ecuId: EcuIdentifier): String = ecuId.value
-    override def from(s: String): Either[ValidationError, EcuIdentifier] = apply(s)
+    override def from(s: String): Either[ValidationError, EcuIdentifier] =
```

Comment:
Not sure it is an improvement. First by removing an `apply` method you allow anyone create an instance of `EcuIdentifier` without a validation. Second, the code is IMO harder to read.

Reply:
Yeah, I can see how this is less readable. But I don't understand how it allows to bypass the validation if `#from` and `#apply` share the same signature 😕.

Reply:
Oh I what you mean now, with the `#apply` form `ValidatedGeneric`. Makes sense, not a good idea then.